### PR TITLE
Animated

### PR DESCRIPTION
**Pull request by mzgoddard**

### CODE REVIEW

File: tests/animated-timer.js [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+  beforeEach(function() {
+    agent = new FakeAgent();
+    fakeTimers = sinon.useFakeTimers();
+  });
+
+  afterEach(function() {
+    fakeTimers.restore();
+  });
+
+  it('creates a timer', function() {
```
Comment (MattSurabian) [inline tests/animated-timer.js:28]: Missing `expect(timer).to.be.an.instanceOf('AnimatedTimer')` or similar

**Comment by MattSurabian** [comment]
This looks good to me, except for that small test issue; I like the latest changes. I have a few thoughts about AnimatedKey that maybe we can talk about but they don't have to delay this PR :+1:

File: src/animated-agent.js [hunk 2]
```diff
@@ -0,0 +1,426 @@
...
+      window.removeEventListener('orientationchange', this.handleResize);
+    }
+  }
+
+  removeAnimatedStyle(animated, animatedEl) {
+    // Return an animated element to the style an animation replaced. This
+    // should return the element to how it was before the animation was played.
+    // This is used at times to return the element to the non animated state to
+    // query the DOM's layout. After use, the style is returned to that of the
+    // any current animation so a user never sees the change.
```
Comment (kadamwhite) [inline src/animated-agent.js:111]: Do we have any intention to auto-generate API docs down the road, e.g. with JSDoc?

Comment (mzgoddard) [inline src/animated-agent.js:111]: Yes that is something I want to do. A lot of the current docs are "internal". But would probably have lots of content cross over.

**Comment by kadamwhite** [comment]
👍 Tests are green and all comments addressed, merging

**Pull request merged**